snow/consensus/snowman/poll/graph_test.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package poll
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/bag"
+)
+
+func TestBuildVotesGraph(t *testing.T) {
+	g := &voteVertex{
+		id: ids.ID{1},
+		descendants: []*voteVertex{
+			{id: ids.ID{11}, descendants: []*voteVertex{
+				{id: ids.ID{111}},
+				{id: ids.ID{112}},
+				{id: ids.ID{113}},
+			}},
+			{id: ids.ID{12}},
+			{id: ids.ID{13}},
+		},
+	}
+	wireParents(g)
+
+	getParent := getParent(g)
+
+	var votes bag.Bag[ids.ID]
+	g.traverse(func(v *voteVertex) {
+		votes.Add(v.id)
+	})
+
+	parents := make(map[ids.ID]ids.ID)
+	children := make(map[ids.ID]*bag.Bag[ids.ID])
+
+	votesGraph := buildVotesGraph(getParent, votes)
```

Comment:
It's not as simple as it seems, since the graph has two way vertices (parent and descendants), I'll have to first define the vertices and then wire the pointers and it will be very verbose :-)